src/http/serializers/request/index.js

```diff
@@ -0,0 +1,112 @@
+import qs from 'qs';
+
+import formatKeyValue from './format.js';
+import { isFile, isArrayOfFile, FileWithData } from './file.js';
+
+function buildFormData(reqForm) {
+  /**
+   * Build a new FormData instance, support array as field value
+   * OAS2.0 - when collectionFormat is multi
+   * OAS3.0 - when explode of Encoding Object is true
+   *
+   * This function explicitly handles Buffers (for backward compatibility)
+   * if provided as a values to FormData. FormData can only handle USVString
+   * or Blob.
+   *
+   * @param {Object} reqForm - ori req.form
+   * @return {FormData} - new FormData instance
+   */
+  return Object.entries(reqForm).reduce((formData, [name, input]) => {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [key, value] of formatKeyValue(name, input, true)) {
+      if (Array.isArray(value)) {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const v of value) {
+          if (ArrayBuffer.isView(v)) {
+            const blob = new Blob([v]);
+            formData.append(key, blob);
+          } else {
+            formData.append(key, v);
+          }
+        }
+      } else if (ArrayBuffer.isView(value)) {
+        const blob = new Blob([value]);
+        formData.append(key, blob);
+      } else {
+        formData.append(key, value);
+      }
+    }
+    return formData;
+  }, new FormData());
+}
+
+// Encodes an object using appropriate serializer.
+function encodeFormOrQuery(data) {
+  /**
+   * Encode parameter names and values
+   * @param {Object} result - parameter names and values
+   * @param {string} parameterName - Parameter name
+   * @return {object} encoded parameter names and values
+   */
+  const encodedQuery = Object.keys(data).reduce((result, parameterName) => {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [key, value] of formatKeyValue(parameterName, data[parameterName])) {
+      if (value instanceof FileWithData) {
+        result[key] = value.valueOf();
+      } else {
+        result[key] = value;
+      }
+    }
+    return result;
+  }, {});
+
+  return qs.stringify(encodedQuery, { encode: false, indices: false }) || '';
+}
+
+// If the request has a `query` object, merge it into the request.url, and delete the object
+// If file and/or multipart, also create FormData instance
+function mergeInQueryOrForm(req = {}) {
```

Comment:
Let's give a proper naming consistent with https://github.com/swagger-api/swagger-js/pull/3482/files#r1576329286. The fact that it's serializing query or form is just an implementation detail, right?

```suggestion
function serializeRequest(req = {}) {
```